packages/issuer/contracts/Issuer.sol

```diff
@@ -227,26 +236,40 @@ contract Issuer is Initializable, Ownable {
         return id;
 	}
 
-	// TO-DO: only Issuer
+    function getMigrationRequestId(uint _certificateId) external onlyOwner returns (uint256) {
```

Comment:
I think that each certificate can have multiple requests to go to public. So for e.g 1 private owner exposes it now while other performs this at later time.

Reply:
True, this should be changed. Thanks.

Reply:
In this case if we want to disallow same migration to be requested. We can store hash of the request (abi.encode(id, owneradd)) or possibly just the  _ownerAddressLeafHash  mapped to certificate id.

mapping(certId => _ownerAddressLeafHash)

Reply:
Also, since this method was lineary growing in the execution time, it would be more and more expensive for users to call. We should if possible to optimize such a O(N) iterations with O(1) mapping calls.